src/xrpld/app/misc/detail/LendingHelpers.cpp

```diff
@@ -978,6 +1124,24 @@ computePaymentComponents(
     };
 }
 
+/* Computes payment components for an overpayment scenario.
+ *
+ * An overpayment occurs when a borrower pays more than the scheduled periodic
+ * payment amount. The overpayment is treated as extra principal reduction,
+ * but incurs a fee and potentially a penalty interest charge.
+ *
+ * The calculation:
+ * 1. Apply the overpayment fee (reduces the effective payment amount)
+ * 2. Calculate penalty interest on the remaining amount
+ * 3. Split the penalty interest into net interest and management fee
+ * 4. Apply the remainder to principal reduction
```

Comment:
Step 1 here is incorrect, is it not? Do you want the comment reflecting the incorrect order and I'll update it when I fix the bug? Or would you rather have the intention documented correctly, the acknowledge that there is a bug to be fixed.

Reply:
Yeah, that's right. I updated to comment to show the correct intent.

Reply:
Fixed.